### PR TITLE
Enhancement #8001: UI data coupling

### LINKED ISSUE #8001: UI & Data Coupling
**Description**
Substitute the static data created in #7998 with the real data in #7999 and handle all states accordingly ( blurred, failed & in progress )

**Dependency**
This module is dependent on #7998 & #7999 to be complete.

**Acceptance Criteria**
- When test is complete and successful, clicking on the dropdown should display the expanded metric data.
  - This action will be tracked with mixpanel
- When tests are complete and successful, clicking on the dropdown next to the global score should display the expanded metric data for each test.
- Users on free plan should see metric data blurred when the test score is blurred and also 'See Report' is disabled.
- Failed or in progress tests should not be expanded
- First test should be expanded by default on RI settings page as long as test is completed and successful
- Clicking on 'See Report' will be tracked in mixpanel.

### PR DESCRIPTION
**Pull request by Miraeld**
# Description

Fixes #8001 
Users now see real GTmetrix performance metrics (LCP, TBT, TTFB, CLS) instead of placeholder data. Each metric is color-coded (green/yellow/red) based on Web Vitals thresholds, providing immediate visual feedback on which areas need optimization. Failed or in-progress tests display appropriate states, ensuring users always understand the status of their performance data.

## Type of change

- [x] Enhancement (non-breaking change which improves an existing functionality).

### How to test

Just make a test, and you should see the new data displayed.

### Affected Features & Quality Assurance Scope

Rocket Insights

### Documentation

This pull request makes several significant changes to the font analysis and visibility logic in `assets/js/wpr-beacon.js`, primarily simplifying and streamlining how elements and fonts are processed. The main improvements include removing complex visibility and styling checks, simplifying external font detection, and making font-face rule extraction synchronous and less error-prone. Additionally, a small change was made in the PHP backend to store metric data as JSON.

**Key changes:**

### JavaScript: Font Analysis and Element Visibility

* Removed the `isElementVisible` and `hasTransparentText` methods from `BeaconUtils`, and replaced their usage with a simplified visibility check that only considers display, visibility, opacity, and element size. This also removes the dependency on checking transparent text properties. [[1]](diffhunk://#diff-da8d2a7d3b6094e8ec5b6029253cd963bf39b7e6db6ae892934a85e4874919eeL27-L85) [[2]](diffhunk://#diff-da8d2a7d3b6094e8ec5b6029253cd963bf39b7e6db6ae892934a85e4874919eeL134-R75) [[3]](diffhunk://#diff-da8d2a7d3b6094e8ec5b6029253cd963bf39b7e6db6ae892934a85e4874919eeL543-R435)
* Removed logic for excluding elements based on tag names (`EXCLUDED_TAG_NAMES`) and the `canElementBeStyledWithFontFamily`/`canElementBeProcessed` methods, simplifying which elements are considered for font analysis to only those above the fold. [[1]](diffhunk://#diff-da8d2a7d3b6094e8ec5b6029253cd963bf39b7e6db6ae892934a85e4874919eeL448-L509) [[2]](diffhunk://#diff-da8d2a7d3b6094e8ec5b6029253cd963bf39b7e6db6ae892934a85e4874919eeL974-L986) [[3]](diffhunk://#diff-da8d2a7d3b6094e8ec5b6029253cd963bf39b7e6db6ae892934a85e4874919eeL999-R730) [[4]](diffhunk://#diff-da8d2a7d3b6094e8ec5b6029253cd963bf39b7e6db6ae892934a85e4874919eeL1192-R923)
* Updated the external font provider detection to use a fixed list of known domains, rather than comparing link origins and applying exclusion lists, streamlining external font stylesheet processing.

### JavaScript: Font-Face Rule Extraction

* Refactored `getFontFaceRules` to be synchronous and much simpler: it now only processes same-origin stylesheets and extracts all font-face rules directly, removing asynchronous fetching, import rule handling, and error-prone recursion.

### JavaScript: Miscellaneous Fixes

* Improved the regex for extracting CSS background URLs to avoid matching whitespace and quotes incorrectly.
* Fixed a potential bug by checking for font family existence in the results before assigning hosted font results.
* Adjusted the logic for above/below the fold font element tracking to only use the above-the-fold check when updating font usage. [[1]](diffhunk://#diff-da8d2a7d3b6094e8ec5b6029253cd963bf39b7e6db6ae892934a85e4874919eeL1110-R841) [[2]](diffhunk://#diff-da8d2a7d3b6094e8ec5b6029253cd963bf39b7e6db6ae892934a85e4874919eeR860-R861)

### PHP: Database Metric Data Storage

* In `inc/Engine/Admin/RocketInsights/Database/Queries/RocketInsights.php`, added logic to store `metric_data` as a JSON-encoded string if present in the test data array.

# Mandatory Checklist

## Code validation

- [x] I validated all the Acceptance Criteria. If possible, provide screenshots or videos.
- [x] I triggered all changed lines of code at least once without new errors/warnings/notices.
- [x] I implemented built-in tests to cover the new/changed code.

## Code style

- [x] I wrote a self-explanatory code about what it does.
- [x] I protected entry points against unexpected inputs.
- [x] I did not introduce unnecessary complexity.
- [x] Output messages (errors, notices, logs) are explicit enough for users to understand the issue and are actionnable.

### CODE REVIEW

File: assets/js/lazyload-css.min.js
Comment (jeawhanlee) [inline assets/js/lazyload-css.min.js:1]: We shouldn't push changes during compilation for this file

File: assets/js/wpr-beacon.js
Comment (jeawhanlee) [inline assets/js/wpr-beacon.js:1]: We shouldn't push changes during compilation for this file

File: assets/js/wpr-beacon.min.js.map
Comment (jeawhanlee) [inline assets/js/wpr-beacon.min.js.map:1]: We shouldn't push changes during compilation for this file

File: inc/Engine/Admin/RocketInsights/Database/Queries/RocketInsights.php [hunk 1]
```diff
@@ -191,4 +192,28 @@ public function unblur_rows() {
...
+
+		// Use table class naming helper for consistency with prefixes.
+		$prefixed_table_name = $this->table_name;
+		// @phpstan-ignore-next-line
+		if ( ! empty( $db->prefix ) ) {
+			$prefixed_table_name = $db->prefix . $this->table_name;
+		}
+
+		return $db->query( "UPDATE `$prefixed_table_name` SET status = 'pending' WHERE status = 'completed'" );
+	}
```
Comment (jeawhanlee) [inline inc/Engine/Admin/RocketInsights/Database/Queries/RocketInsights.php]: What is the use of this method here?
From what I see, this method is defined somewhere else in this [PR](https://github.com/wp-media/wp-rocket/pull/8009/)

File: src/js/global/ajax.js [hunk 2]
```diff
@@ -770,14 +770,26 @@ document.addEventListener('DOMContentLoaded', function() {
...
+			var isCurrentlyVisible = $details.is(':visible');
+			
+			// Track expand/collapse if Mixpanel is available.
+			if (window.rocket_mixpanel_data && window.rocket_mixpanel_data.optin_enabled && window.mixpanel) {
+				window.mixpanel.track('Rocket Insights Metrics Expanded', {
+					context: 'wp_plugin',
+					row_id: insightsId,
+					action: isCurrentlyVisible ? 'collapse' : 'expand'
+				});
+			}
```
Comment (jeawhanlee) [inline src/js/global/ajax.js]: We can move this to the `toggleSingleRowVisibility` function, by doing that, we'll not need to redefine selectors. Also, we don't need to track the collapse action, only the expand. See [here](https://group-onecom.slack.com/archives/C08EFCYRQ2X/p1770284235465259?thread_ts=1770284023.599999&cid=C08EFCYRQ2X)

File: src/js/global/ajax.js [hunk 3]
```diff
@@ -796,4 +808,14 @@ document.addEventListener('DOMContentLoaded', function() {
 			updateRowStylingForLastItem($selectors);
 		});
 	}
+
+	// Track "See Report" clicks in Rocket Insights.
+	$(document).on('click', '.wpr-ri-report', function(e) {
+		// Only track if link is not disabled and mixpanel is available.
+		if (!$(this).hasClass('wpr-ri-action--disabled') && window.rocket_mixpanel_data && window.rocket_mixpanel_data.optin_enabled && window.mixpanel) {
+			window.mixpanel.track('Rocket Insights See Report', {
+				context: 'wp_plugin'
```
Comment (jeawhanlee) [inline src/js/global/ajax.js]: Shoudln't we also pass the test id?

File: views/settings/partials/rocket-insights/table-row.php [hunk 4]
```diff
@@ -187,24 +191,97 @@
...
-							<div class="metric-value ri-warning">
-								<p>1.4s</p>
-							</div>
-							<div class="metric-value ri-success">
-								<p>700ms</p>
-							</div>
-							<div class="metric-value ri-warning">
-								<p>0.145s</p>
-							</div>
+					<div class="row-right <?php echo esc_attr( $rocket_ri_blurred ); ?>">
```
Comment (jeawhanlee) [inline views/settings/partials/rocket-insights/table-row.php]:
```suggestion
					<div class="row-right">
```

File: views/settings/partials/rocket-insights/table-row.php [hunk 5]
```diff
@@ -187,24 +191,97 @@
...
+								],
+								[
+									'key'   => 'time_to_first_byte',
+									'value' => $rocket_metric_data['time_to_first_byte'] ?? null,
+								],
+								[
+									'key'   => 'cumulative_layout_shift',
+									'value' => $rocket_metric_data['cumulative_layout_shift'] ?? null,
+								],
+							];
```
Comment (jeawhanlee) [inline views/settings/partials/rocket-insights/table-row.php]: Can we separate this kinds of logic away from the view. WDYT?

**Review by jeawhanlee — CHANGES_REQUESTED** [review]

**Comment by Miraeld** [comment]
### PR Feedback Addressed

| Feedback Item | Resolution |
|--------------|------------|
| **Remove compiled files from PR** | Reverted compiled files to base branch (`lazyload-css.min.js`, `wpr-beacon.min.js.map`, etc.) |
| **Unused PHP tracking methods** | Verified - tracking is JS-only, no PHP methods to remove |
| **Move Mixpanel tracking** | Moved into `toggleSingleRowVisibility()` function |
| **Only track expand, not collapse** | Now only tracks when showing (expand), not hiding (collapse) |
| **Pass test ID to "See Report"** | Added `row_id: insightsId` to Mixpanel tracking |
| **Move metric logic out of view** | Created `MetricFormatter` class with threshold constants and formatting methods |
| **Remove duplicate blurred class** | Removed duplicate `$rocket_ri_blurred` from `row-right` div |
| **Check duplicate method with PR #8009** | Not a duplicate - `parse_metric_data()` parses JSON from DB, `get_formatted_metrics()` formats for display |

### Additional Updates

- Merged latest `feature/ri-advanced-indicators` to stay up-to-date with base branch

### Files Changed

- **New**: `inc/Engine/Admin/RocketInsights/MetricFormatter.php` - Centralized metric formatting logic
- **Modified**: `inc/Engine/Admin/RocketInsights/Render.php` - Added MetricFormatter dependency injection
- **Modified**: `inc/Engine/Admin/RocketInsights/ServiceProvider.php` - Registered `ri_metric_formatter`
- **Modified**: `src/js/global/ajax.js` - Fixed Mixpanel tracking placement and behavior
- **Modified**: `views/settings/partials/rocket-insights/table-row.php` - Simplified to use pre-formatted metrics

File: src/js/global/ajax.js [hunk 6]
```diff
@@ -732,12 +732,33 @@ document.addEventListener('DOMContentLoaded', function() {
 			$details.show('fast');
 			$img.attr('src', carets.down);
 			
+			// Track expand only expand metric action.
+			handleMetricActionTracking('expand', insightsId);
```
[the remark this answers is not in the document]
Comment (jeawhanlee) [reply src/js/global/ajax.js:736]: The backend already caters for checking for in-progress or failed state. This scenario will never happen.
